crates/torii/core/src/engine.rs

```diff
@@ -191,26 +214,21 @@ impl<'db, P: Provider + Sync> Engine<'db, P> {
                 let event_id =
                     format!("0x{:064x}:0x{:04x}:0x{:04x}", block.block_number, tx_idx, event_idx);
 
-                Self::process_event(self, &block, &invoke_receipt, &event_id, event).await?;
+                Self::process_event(self, block, &receipt, &event_id, event).await?;
             }
 
             if world_event {
                 let transaction_id = format!("0x{:064x}:0x{:04x}", block.block_number, tx_idx);
 
-                Self::process_transaction(
-                    self,
-                    &block,
-                    &invoke_receipt,
-                    &transaction_id,
-                    invoke_transaction,
-                )
-                .await?;
+                Self::process_transaction(self, block, &receipt, &transaction_id, transaction)
+                    .await?;
             }
-        }
 
-        info!("processed block: {}", block.block_number);
-
-        Ok(())
+            Ok(())
+        } else {
+            warn!("transaction receipt not found: {}", transaction_hash);
```

Comment:
In this case, we should probably retry processing this block again

Reply:
No, I don't think that we should outrightly. The line also gets called when the receipt is not of `L1Handler` and `Invoke` txn types. 

I'll update the code to make this distinction and retry for failed fetch

Reply:
Made some changes to retry failed transaction processing